src/main/kotlin/de/cyface/collector/storage/cloud/MongoDatabase.kt

```diff
@@ -78,12 +77,10 @@ class MongoDatabase(private val mongoClient: MongoClient, private val collection
     override fun exists(deviceIdentifier: String, measurementIdentifier: Long): Future<Boolean> {
         val ret = Promise.promise<Boolean>()
         val query = JsonObject()
-            .put("metadata.${FormAttributes.DEVICE_ID.value}", deviceIdentifier)
-            .put("metadata.${FormAttributes.MEASUREMENT_ID.value}", measurementIdentifier.toString())
+            .put("properties.${FormAttributes.DEVICE_ID.value}", deviceIdentifier)
+            .put("properties.${FormAttributes.MEASUREMENT_ID.value}", measurementIdentifier.toString())
             // Ensure we don't interpret attachments as measurements
-            .put("metadata.${FormAttributes.ATTACHMENT_ID.value}", JsonObject().put("\$exists", false))
-        // query.put("features.0.properties.${FormAttributes.DEVICE_ID.value}", deviceIdentifier)
-        // query.put("features.0.properties.${FormAttributes.MEASUREMENT_ID.value}", measurementIdentifier)
+            .put("properties.${FormAttributes.ATTACHMENT_ID.value}", JsonObject().put("\$exists", false))
```

Comment:
The identifiers are stored as string in the database, see above.
The measurement/attachment metadata is stored as a Feature-geometry (not a Feature Collection), see above.